src/deadline_worker_agent/sessions/actions/run_attachment_download.py

```diff
@@ -0,0 +1,338 @@
+import sysconfig
+from typing import Any, TYPE_CHECKING, Optional
+from dataclasses import asdict
+
+from deadline.job_attachments.asset_manifests import BaseAssetManifest
+from deadline.job_attachments.models import (
+    Attachments,
+    PathFormat,
+    JobAttachmentS3Settings,
+    ManifestProperties,
+    PathMappingRule,
+    JobAttachmentsFileSystem,
+)
+from deadline.job_attachments.os_file_permission import (
+    FileSystemPermissionSettings,
+    PosixFileSystemPermissionSettings,
+    WindowsFileSystemPermissionSettings,
+    WindowsPermissionEnum,
+)
+
+from openjd.sessions import (
+    LOG as OPENJD_LOG,
+    PathMappingRule as OpenjdPathMapping,
+    PosixSessionUser,
+    WindowsSessionUser,
+)
+from openjd.model.v2023_09 import (
+    EmbeddedFileTypes as EmbeddedFileTypes_2023_09,
+    EmbeddedFileText as EmbeddedFileText_2023_09,
+    Action as Action_2023_09,
+    StepScript as StepScript_2023_09,
+    StepActions as StepActions_2023_09,
+)
+from openjd.model import ParameterValue
+
+from ..session import Session
+from ...log_messages import SessionActionLogKind
+
+
+from .openjd_action import OpenjdAction
+
+if TYPE_CHECKING:
+    from concurrent.futures import Future
+    from ..session import Session
+    from ..job_entities import JobAttachmentDetails, StepDetails
+
+
+logger = getLogger(__name__)
+
+
+class SyncCanceled(Exception):
+    """Exception indicating the synchronization was canceled"""
+
+    pass
+
+
+class AttachmentDownloadAction(OpenjdAction):
+    """Action to synchronize input job attachments for a AWS Deadline Cloud job
+
+    Parameters
+    ----------
+    id : str
+        The unique action identifier
+    """
+
+    _future: Future[None]
+    _job_attachment_details: Optional[JobAttachmentDetails]
+    _step_details: Optional[StepDetails]
+    _step_script: Optional[StepScript_2023_09]
+
+    def __init__(
+        self,
+        *,
+        id: str,
+        session_id: str,
+        job_attachment_details: Optional[JobAttachmentDetails] = None,
+        step_details: Optional[StepDetails] = None,
+    ) -> None:
+        super(AttachmentDownloadAction, self).__init__(
+            id=id,
+            action_log_kind=(
+                SessionActionLogKind.JA_SYNC
+                if step_details is None
+                else SessionActionLogKind.JA_DEP_SYNC
+            ),
+            step_id=step_details.step_id if step_details is not None else None,
+        )
+        self._job_attachment_details = job_attachment_details
+        self._step_details = step_details
+        self._logger = LoggerAdapter(OPENJD_LOG, extra={"session_id": session_id})
+
+    def set_step_script(self, manifests, path_mapping, s3_settings) -> None:
+        profile = os.environ.get("AWS_PROFILE")
+        deadline_path = os.path.join(Path(sysconfig.get_path("scripts")), "deadline")
+
+        script = "#!/usr/bin/env bash\n\n{} attachment download -m {} --path-mapping-rules {} --s3-root-uri {}  --profile {}".format(
```

Comment:
nit, wdyt if the script line was a formatted multi-line string?

script = """
#!/usr/bin/env bash {} \
    attachment download \
    -m {} \
    --path-mapping-rules {} \
    --s3-root-uri {} \
    --profile {}"""

Reply:
I tired it, and the test started to fail. I wouldn't worry about it too much since it will be replaced by a python embedded file.